corehq/form_processor/tests/test_reindex_accessors.py

```diff
@@ -0,0 +1,202 @@
+import time
+import uuid
+from datetime import datetime
+from unittest import SkipTest
+
+from django.conf import settings
+from django.test import TestCase
+
+from corehq.form_processor.backends.sql.dbaccessors import (
+    CaseAccessorSQL, FormReindexAccessor, CaseReindexAccessor,
+    LedgerAccessorSQL, LedgerReindexAccessor
+)
+from corehq.form_processor.models import LedgerValue, CommCareCaseSQL
+from corehq.form_processor.tests import FormProcessorTestUtils, PartitionConfig, create_form_for_test
+
+
+class BaseReindexAccessorTest(object):
+    accessor_class = None
+
+    @classmethod
+    def setUpClass(cls):
+        super(BaseReindexAccessorTest, cls).setUpClass()
+        cls.domain = uuid.uuid4().hex
+        # since this test depends on the global form list just wipe everything
+        FormProcessorTestUtils.delete_all_sql_forms()
+        FormProcessorTestUtils.delete_all_v2_ledgers()
+        FormProcessorTestUtils.delete_all_sql_cases()
+
+        cls.first_batch = cls._get_doc_ids(cls._create_docs(4))
+        cls.middle = datetime.utcnow()
+        time.sleep(.02)
```

Comment:
nit: actually patching the `datetime` object could make this more reliable for different people's computer. there are some examples in other tests here: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports/tests/test_daterange.py#L31